src/python_testing/matter_testing_support.py

```diff
@@ -707,6 +709,19 @@ def check_pics(self, pics_key: str) -> bool:
         pics_key = pics_key.strip().upper()
         return pics_key in picsd and picsd[pics_key]
 
+    def openCommissioningWindow(self, dev_ctrl: ChipDeviceCtrl, node_id: int) -> CommissioningParameters:
+        rnd_discriminator = random.randint(0, 4095)
+        try:
+            commissioning_params = dev_ctrl.OpenCommissioningWindow(nodeid=node_id, timeout=900, iteration=1000,
+                                                                    discriminator=rnd_discriminator, option=1)
+            discriminator_dict = {"random_discriminator": rnd_discriminator}
+            params = (commissioning_params, discriminator_dict)
+
+            return params
+
+        except InteractionModelError as e:
+            asserts.fail(e.status, 'Failed to open commissioning window')
```

Comment:
I think you still need to return something even if you assert failure - you only need to not return if you throw an exception.